src/argilla/server/contexts/datasets.py

```diff
@@ -746,35 +737,72 @@ async def _build_record_vectors(
     record_schema: Union["RecordCreate", "RecordUpdate", "RecordUpdateWithId"],
     build_vector_func: Callable[[List[float], UUID], VectorClass],
     cache: Dict[str, VectorSettingsSchema] = {},
-    record_position: Optional[int] = None,
-) -> List[VectorClass]:
+) -> Tuple[List[VectorClass], Dict[str, VectorSettingsSchema]]:
     """Create vectors for a record."""
     if not record_schema.vectors:
-        return []
+        return [], cache
 
     vectors = []
     for vector_name, vector_value in record_schema.vectors.items():
         try:
-            await _validate_vector(db, dataset.id, vector_name, vector_value, vectors_settings=cache)
+            cache = await _validate_vector(db, dataset.id, vector_name, vector_value, vectors_settings=cache)
             vectors.append(build_vector_func(vector_value, cache[vector_name].id))
         except ValueError as e:
-            err_msg = (
-                f"Provided vector with name={vector_name} of record at position {record_position} is not valid: {e}"
-                if record_position is not None
-                else f"Provided vector with name={vector_name} is not valid: {e}"
-                if record_position is not None
-                else f"Provided vector with name={vector_name} is not valid: {e}"
-            )
-            raise ValueError(err_msg) from e
-
-    return vectors
+            raise ValueError(f"vector with name={vector_name} is not valid: {e}") from e
+    return vectors, cache
 
 
 async def _exists_records_with_ids(db: "AsyncSession", dataset_id: UUID, records_ids: List[UUID]) -> List[UUID]:
     result = await db.execute(select(Record.id).filter(Record.dataset_id == dataset_id, Record.id.in_(records_ids)))
     return result.scalars().all()
 
 
+async def _update_record(
+    db: "AsyncSession", dataset: Dataset, record_update: "RecordUpdateWithId", caches: Optional[Dict[str, Any]] = None
+) -> Tuple[Dict[str, Any], List[Suggestion], List[VectorSchema], bool, Dict[str, Any]]:
+    if caches is None:
+        caches = {
+            "metadata_properties": {},
+            "questions": {},
+            "vector_settings": {},
+        }
+
+    params = record_update.dict(exclude_unset=True)
+    needs_search_engine_update = False
```

Comment:
Do we really need this? It's a little bit error-prone (like forgetting set it to `True` and then not indexing). I would always reindex after an update instead.

Reply:
I was doing it like this to avoid querying all the records with their vectors from the database and then call the `index_records` with all of them as this can be a bit heavy operation, right?

Reply:
I see. Let's continue with that then, we can try to refactor this in a near future. Thanks for the explanation.